tools/releaser.sh

```diff
@@ -0,0 +1,210 @@
+
+dir=$(dirname "$0")
+
+# Default values for the arguments.
+GH_ORG="noobaa"
+GH_REPO="noobaa-core"
+OCI_ORG="noobaa"
+DRY_RUN="false"
+SYNC_OPERATOR_REPOSITORY=""
+
+# Default values for the environment variables.
+GITHUB_TOKEN=${GITHUB_TOKEN:-}
+DOCKERHUB_USERNAME=${DOCKERHUB_USERNAME:-}
+DOCKERHUB_TOKEN=${DOCKERHUB_TOKEN:-}
+QUAY_USERNAME=${QUAY_USERNAME:-}
+QUAY_TOKEN=${QUAY_TOKEN:-}
+BASE_BRANCH=${BASE_BRANCH:-master}
+
+function check_environment_variables() {
+  if [[ -z "${GITHUB_TOKEN}" ]]; then
+    echo "GITHUB_TOKEN environment variable is not set."
+    exit 1
+  fi
+  if [[ -z "${DOCKERHUB_USERNAME}" ]]; then
+    echo "DOCKERHUB_USERNAME environment variable is not set."
+    exit 1
+  fi
+  if [[ -z "${DOCKERHUB_TOKEN}" ]]; then
+    echo "DOCKERHUB_TOKEN environment variable is not set."
+    exit 1
+  fi
+  if [[ -z "${QUAY_USERNAME}" ]]; then
+    echo "QUAY_USERNAME environment variable is not set."
+    exit 1
+  fi
+  if [[ -z "${QUAY_TOKEN}" ]]; then
+    echo "QUAY_TOKEN environment variable is not set."
+    exit 1
+  fi
+}
+
+function usage() {
+  echo "Usage: $0 [options]"
+  echo "Options:"
+  echo "  --help - Print this help message."
+  echo "  --oci-org <org> - The organization of the OCI images."
+  echo "  --gh-org <org> - The organization of the GitHub repository."
+  echo "  --gh-repo <repo> - The GitHub repository name."
+  echo "  --dry-run - Do not create the release."
+  echo "  --sync-operator-repository <repository> - The operator repository to sync."
+}
+
+function parse_args() {
+  while [ "$#" -gt 0 ]; do
+    case "$1" in
+    --gh-org)
+      GH_ORG="$2"
+      shift
+      ;;
+    --gh-repo)
+      GH_REPO="$2"
+      shift
+      ;;
+    --oci-org)
+      OCI_ORG="$2"
+      shift
+      ;;
+    --dry-run)
+      DRY_RUN="true"
+      ;;
+    --sync-operator-repository)
+      SYNC_OPERATOR_REPOSITORY="$2"
+      shift
+      ;;
+    -h | --help)
+      usage
+      exit 0
+      ;;
+    *)
+      usage
+      exit 1
+      ;;
+    esac
+    shift
+  done
+}
+
+function strip_prefix() {
+  local str="$1"
+  local prefix="$2"
+
+  echo ${str#"$prefix"}
+}
+
+function version_without_v() {
+  strip_prefix "$1" v
+}
+
+function version_with_v() {
+  local without_v=$(version_without_v "$1")
+
+  echo "v$without_v"
+}
+
+# get_noobaa_version returns the NooBaa version by reading the `package.json` file.
+#
+# Note: If the second argument is set to 1, it will append "v" to the version while no
+# argument will return the version without "v"
+function get_noobaa_version() {
+  local version=$(npm -s run env echo '$npm_package_version')
+  if [[ $# -gt 1 ]]; then
+    echo "[Error]: expecting one argument at the most"
+  fi
+  if [[ $# -eq 1 ]]; then
+    if [[ "$1" == "1" ]]; then
+      version_with_v "$version"
+    else
+      version_without_v "$version"
+      echo "[Warn]: Unknown argument passed to get_noobaa_version."
+    fi
+  else
+    version_without_v "$version"
+  fi
+}
```

Comment:
follow the get_noobaa_version in the operator.

Reply:
Actually this is from the operator.

Reply:
So I missed it on the operator as well.
We cant use `echo "[Error]:` or `echo "[Warn]:` here unless we are handling this error on the call.
Currently, upon error or this warning, we will change the version to include this string.